mapping_parenting_tech/analysis/prototyping/play_store/fn_tester.py

```diff
@@ -0,0 +1,140 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: -all
+#     comment_magics: true
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.13.6
+#   kernelspec:
+#     display_name: Python 3.8.0 ('mapping_parenting_tech')
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# # Testing notebook
+# Testing refactoring of functions to manage ids, details and reviews of Play Store apps
+
+# %%
+from mapping_parenting_tech.utils import play_store_utils as psu
+from mapping_parenting_tech import PROJECT_DIR, logging
+
+# %% [markdown]
+# ## Working with app ids
+# ### Getting app ids from source/s
+# App ids are retrieved in two ways:
+# 1. parsing an HTML file saved from a Play Store web page - this can be done with a single web page or via a folder that contains multiple web pages
+# 2. snowballing from a given app id and using that to identify related apps
+
+# %%
+# test downloading apps from a single HTML web page
+appIds = psu.get_playstore_app_ids_from_html("play_store/parenting_top_free.html")
+appIds
+
+# %%
+# test parsing a folder of HTML files
+appIds = psu.get_playstore_app_ids_from_folder("play_store/education_apps")
+len(appIds)
+
+# %%
+# test retrieving related apps to a depth of 3
+appIds = psu.app_snowball("uk.org.bestbeginnings.babybuddymobile", 3)
+len(appIds)
+
```

Comment:
Got an error here. I guess your snowball routine should be adjusted to handle a case when there are no related apps...
```
2022-03-22 10:49:10,204 - root - WARNING - uk.org.bestbeginnings.babybuddymobile had a problem. Maybe it has no related apps.
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/var/folders/7_/xc8h4x6544z_85zlhcsnmtv40000gp/T/ipykernel_43180/1692066187.py in <module>
      1 # test retrieving related apps to a depth of 3
----> 2 appIds = psu.app_snowball("uk.org.bestbeginnings.babybuddymobile", 3)
      3 len(appIds)

~/Documents/mapping_parenting_tech/mapping_parenting_tech/utils/play_store_utils.py in app_snowball(seed_app_id, depth, __current_depth)
    192 
    193     if __current_depth < depth:
--> 194         for this_app in similar_apps:
    195             snowball.update(app_snowball(this_app, depth, (__current_depth + 1)))
    196 

TypeError: 'NoneType' object is not iterable
```